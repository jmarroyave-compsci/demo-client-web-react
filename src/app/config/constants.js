import _package_ from 'package.json';

const Constants = {
	DEBUG : process.env.NEXT_PUBLIC__DEBUG,
	LOCAL : process.env.NEXT_PUBLIC__LOCAL,
	APP : {
		BASE_PATH: process.env.NEXT_PUBLIC__SITE_URL,
		SITE_URL: process.env.NEXT_PUBLIC__SITE_URL,
		VERSION : _package_.version,
		TITLE: process.env.NEXT_PUBLIC__TITLE,
		PAGE_TITLE: process.env.NEXT_PUBLIC__PAGE_TITLE
	},
	SERVERS: {
		DATA: process.env.NEXT_PUBLIC__DATA, 
		GRAPHQL :  process.env.NEXT_PUBLIC__DATA_GRAPHQL,
	},
	DEFAULTS : {
		PAGING_PAGE_SIZE: 10,
	},
	PLUGINS : {
		GOOGLE: {
			ANALYTICS: process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS,
			MAPS: process.env.NEXT_PUBLIC__GOOGLE_MAPS_KEY,
		},
	},
	FEATURES : {
		SEARCH : (process.env.NEXT_PUBLIC__FEATURE_SEARCH == "true"),
	},
}

export default Constants;