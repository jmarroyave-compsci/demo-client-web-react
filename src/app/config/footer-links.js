import _links from './links';

const links = [
  { "name" : "jmarroyave", items: [
    { "name": "blog", "link" : _links.blog },
  ] },
  { "name" : "labs", items: [
    { "name": "stackoverflow", "link" : _links.labs_trends },
    { "name": "mlb", "link" : _links.labs_mlb },
    { "name": "search", "link" : _links.labs_search },
  ] },
  { "name" : "knowledge base", items: [
    { "name": "android", "link" : _links.kb_android },
  ] },
];

export default links;