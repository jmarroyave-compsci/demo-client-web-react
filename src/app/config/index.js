import _package_ from 'package.json';

const Constants = {
	TEST: (process.env.TESTING === "true") ? true : false,
	DEBUG : process.env.NEXT_PUBLIC__DEBUG,
	LOCAL : process.env.NEXT_PUBLIC__LOCAL,
	APP : {
		BASE_PATH: process.env.NEXT_PUBLIC__BASE_PATH,
		SITE_URL: process.env.NEXT_PUBLIC__SITE_URL,
		VERSION : _package_.version,
		TITLE: process.env.NEXT_PUBLIC__TITLE,
		DESCRIPTION: process.env.NEXT_PUBLIC__DESCRIPTION,
		PAGE_TITLE: process.env.NEXT_PUBLIC__PAGE_TITLE,
		HOME_URL: `${process.env.NEXT_PUBLIC__SITE_URL}${process.env.NEXT_PUBLIC__BASE_PATH}`,
	},
	SERVERS: {
		DATA: process.env.NEXT_PUBLIC__SERVER, 
		WEB_SOCKETS: process.env.NEXT_PUBLIC__SERVER, 
		GRAPHQL :  `${process.env.NEXT_PUBLIC__SERVER}/${process.env.NEXT_PUBLIC__SERVER_API_VERSION}/graphql`,
	},
	DEFAULTS : {
		PAGING_PAGE_SIZE: 10,
		IMAGE_PLACEHOLDER: "/img/ph.jpg",
	},
	PLUGINS : {
		GOOGLE: {
			ANALYTICS: process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS,
			MAPS: process.env.NEXT_PUBLIC__GOOGLE_MAPS_KEY,
		},
	},
	FEATURES : {
		SEARCH : (process.env.NEXT_PUBLIC__FEATURE_SEARCH == "true"),
	},
	SOCIAL_NETWORKS: {
		CARD : {
			TITLE : `${process.env.NEXT_PUBLIC__TITLE} v.${_package_.version}`,
			THUMBNAIL_TWITTER: `${process.env.NEXT_PUBLIC__BASE_PATH}${process.env.NEXT_PUBLIC__CARD_THUMBNAIL_TWITTER}`,
		},
		TWITTER: process.env.NEXT_PUBLIC__SOCIAL_NETWORKS_TWITTER,
	}
}

export default Constants;