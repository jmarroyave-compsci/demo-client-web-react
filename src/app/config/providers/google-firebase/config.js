const config = {
  apiKey: "AIzaSyDaNheMMUtsqx8AjiJAWDcfXwAEBYLJUVY",
  authDomain: "labs-f885b.firebaseapp.com",
  projectId: "labs-f885b",
  //projectId: "labs-f885",
  storageBucket: "labs-f885b.appspot.com",
  messagingSenderId: "606766444808",
  appId: "1:606766444808:web:41eec12c129d582727bcbd",
  measurementId: "G-TNBYCW57TS"
};

export default config;