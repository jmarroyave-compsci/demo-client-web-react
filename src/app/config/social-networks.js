const links = [
  /*
	{enable: true, url: 'https://plus.google.com/112618720945759819017', icon: 'google-plus', color: '#d34836'},
	{enable: true, url: 'https://twitter.com/jmarroyave', icon: 'twitter', color: '#00aced'},
	*/
  { enable: true, url: 'https://facebook.com/jmarroyave', icon: 'facebook', color: '#3b5998' },
  { enable: true, url: 'https://www.linkedin.com/in/jmarroyave', icon: 'linkedin', color: '#0077B5' },
  { enable: true, url: 'https://github.com/jmarroyave', icon: 'github', color: '#000' },
  { enable: true, url: 'mailto:jmarroyave.compsci@gmail.com', icon: 'envelope-o', color: '#000' },
];


export default links;