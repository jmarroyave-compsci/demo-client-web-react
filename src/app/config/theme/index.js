import React from 'react';
import { createTheme } from '@mui/material/styles';
import { red } from '@mui/material/colors';

// Create a theme instance.
export const getTheme = ( mode ) => {
  return createTheme({
    palette: {
      mode: mode,
      primary: {
        main: '#556cd6',
      },
      secondary: {
        main: '#19857b',
      },
      error: {
        main: red.A400,
      },
    },
  });
}