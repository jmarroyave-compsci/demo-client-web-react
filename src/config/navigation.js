
class Navigation {
  constructor(){
    this.drawer = {
      show: false,
    };

    this.options = [
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Home', 
        icon: 'home', 
        route: '/', 
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Movies', 
        icon: 'movie', 
        route: '/movies', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Movie', 
          icon: 'movie', 
          route: '/movie/:id/:name', 
        },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'Podcasts', 
        icon: 'podcasts', 
        route: '/podcasts', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Podcast', 
          icon: 'podcast', 
          route: '/podcast/:id/:name', 
        },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'People', 
        icon: 'people', 
        route: '/people', 
      },
        {
          showInMenu: false, 
          showInDrawer: false, 
          name: 'Person', 
          icon: 'person', 
          route: '/person/:id/:name', 
        },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Awards', 
        route: '/stories/awards/:entity?/:year?/:page?', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Stories', 
        route: '/stories', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'PodcastMusic', 
        route: '/podcasts/stories/music/:page?', 
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        separator: true,
      },
      {
        showInMenu: true, 
        showInDrawer: true, 
        name: 'About', 
        icon: 'info', 
        route: '/about', 
      },
      {
        showInMenu: false, 
        showInDrawer: false, 
        name: 'Search', 
        icon: 'search', 
        route: '/search/:qry/:page?', 
      },
    ];
  }
}



const nav = new Navigation();
export default nav;