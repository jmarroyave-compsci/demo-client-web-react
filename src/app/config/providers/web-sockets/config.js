import config from 'app/config';

const CONF = {
  SERVER: config.SERVERS.WEB_SOCKETS
};

export default CONF;