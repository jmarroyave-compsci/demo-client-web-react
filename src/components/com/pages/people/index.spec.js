import React from 'react';
import {render, fireEvent, waitFor, screen, prettyDOM} from '@testing-library/react'
import '@testing-library/jest-dom'

import { itMustHaveNoErrors, TestProvider } from "tests/jest/shared"

import itemState from './automata/tests/item.json'
import TestComponent from '.';
import config from './.config';

describe(`${config.automata.name.toUpperCase()}`, () => {
  beforeEach(() => { 
  }); 
 
  it('compare Page against snapshot', () => {
    const renderer = render(
      <TestProvider state={itemState}>
        <TestComponent render="page"/>
      </TestProvider>
    )

    itMustHaveNoErrors(renderer)
    //expect(renderer.container).toMatchSnapshot();
  });

  it('compare against snapshot', () => {
    const renderer = render(
      <TestProvider state={itemState}>
        <TestComponent render="banner"/>
      </TestProvider>
    )

    itMustHaveNoErrors(renderer)
    expect(renderer.container).toMatchSnapshot();
  });
});

