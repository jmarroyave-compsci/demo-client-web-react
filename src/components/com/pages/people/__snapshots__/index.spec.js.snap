// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directors story compare against snapshot 1`] = `
<div>
  <div />
</div>
`;
